### ISSUE TITLE
None of this works

### ISSUE THREAD

**Issue by KubaO** [MicrosoftDocs/visualstudio-docs#3812]
None of those targets work in C# projects (.csproj) in latest VS 2019.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 868d865b-89d9-ec87-d569-862d168dc6ca
* Version Independent ID: a95f34e2-8f4b-4a4b-cb7e-075e4ceebd69
* Content: [Extend the build process - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-extend-the-visual-studio-build-process?view=vs-2019#feedback)
* Content Source: [docs/msbuild/how-to-extend-the-visual-studio-build-process.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/msbuild/how-to-extend-the-visual-studio-build-process.md)
* Product: **visual-studio-windows**
* Technology: **msbuild**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by msebolt** [comment]
@KubaO  Thank you for the feedback.  We're sorry to hear you're experiencing issues.

Have you tried the **Clean Solution** option under the **Build** menu?  The **Clean** option is specifically meant for instances when all targets need to be rebuilt.

**Comment by KubaO** [comment]
Of course cleaning will "fix it", but do you seriously expect someone to clean before every single build, just so that the updated source files will rebuild the generated targets? C# output is also "generated", and the dependencies are properly tracked and changing source files will cause recompilation. It's natural to assume that custom build targets, with properly specified `Inputs` and `Outputs`, will be included in msbuild's dependency resolution.

The thing is: this simplistic documentation is a lie to children. It takes a bit more to have the dependencies picked up, and it's not documented anywhere.

**Comment by msebolt** [comment]
@KubaO  You're correct in that the Visual Studio IDE portion referenced above represents only the basic core (or "shell") concepts of the MSBuild engine.  

Perhaps you can refer to the [MSBuild](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild?view=vs-2019) documentation for the more advanced concepts?

**Comment by Science-and-Motion-Sports** [comment]
@KubaO I had the same issue. I noticed that everything was fine when I ran msbuild.exe from the command line. The reason for that is that Visual Studio contains its own (faulty) logic to determine if anything changed and if it thinks nothing changed then Visual Studio does not even run msbuild.exe. The solution is to disable this logic and always run msbuild.
To do that, add the following to your project file:
```
  <PropertyGroup>
    <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
  </PropertyGroup>
```
This property is documented here: https://docs.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2019
